src/interface/state_descriptor.hpp

```diff
@@ -245,7 +245,34 @@ class StateDescriptor {
   Dictionary<Dictionary<Metadata>> swarmValueMetadataMap_;
 };
 
-using Packages_t = Dictionary<std::shared_ptr<StateDescriptor>>;
+class Packages_t {
+ public:
+  Packages_t() = default;
+  bool Add(const std::shared_ptr<StateDescriptor> &package) {
+    const auto &name = package->label();
+    if (packages_.count(name) == 0) {
+      packages_[name] = package;
+      return true;
+    }
+    return false;
+  }
+  std::shared_ptr<StateDescriptor> &Get(const std::string &name) {
```

Comment:
I feel like this should return `std::shared_ptr<StateDescriptor> const &`. Remember - `const` in this case would only confer _shallow_ immutability. So it would prevent people from changing the slot to a wholly different `shared_ptr<StateDescriptor>`, but still allow them to make changes to the existing `StateDescriptor`. Of course, this doesn't do much because you could then just mutate the `StateDescriptor` to violate invariants, too. Idk, we really ought to bifurcate our public API and internal implementation details more, but that's a discussion for another time.

Reply:
Done.